rust/src/lib.rs

```diff
@@ -55,72 +55,64 @@
 mod colour;
 pub use colour::Colour;
 
+mod palette;
+pub use palette::Palette;
+
 mod flavour;
 pub use flavour::Flavour;
 
-mod latte;
-pub use latte::LATTE;
-
-mod frappe;
-pub use frappe::FRAPPE;
-
-mod macchiato;
-pub use macchiato::MACCHIATO;
-
-mod mocha;
-pub use mocha::MOCHA;
-
 /// Returns an iterator over the four delicious Catppuccin flavours.
 pub fn flavours_iter() -> std::array::IntoIter<Flavour, 4> {
-    [LATTE, FRAPPE, MACCHIATO, MOCHA].into_iter()
+    [Flavour::Latte, Flavour::Frappe, Flavour::Macchiato, Flavour::Mocha].into_iter()
 }
```

Comment:
i think this should now just be an `iter` method on `Flavour`, what do you think?

Reply:
I agree with @backwardspy here, though the function should be called `into_iter()` since it's returning an `IntoIter` type, see [C-ITER-TY in the API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#iterator-type-names-match-the-methods-that-produce-them-c-iter-ty)

Reply:
`into_iter` as suggested by @nyxkrage